### PR TITLE
Fix a bug with unnamed containers (normal client)

### PR DESCRIPTION
**Pull request by Zbizu**
**steps to reproduce:**
1. remove item that is a container from items.xml
2. log in and open that container

**expected result:** it opens without label like it does on otclient
**observed result:** client crash with log pointing to container without name

reproduction rate: 100%

**what this fix does:**
it labels unnamed containers as item of type xxxx (number)

### CODE REVIEW

File: src/protocolgame.cpp [hunk 1]
```diff
@@ -1484,7 +1484,15 @@ void ProtocolGame::sendContainer(uint8_t cid, const Container* container, bool h
 		msg.addString("Browse Field");
 	} else {
 		msg.addItem(container);
-		msg.addString(container->getName());
```
Comment (nekiro) [inline src/protocolgame.cpp:1487]:
```
const std::string& containerName = container->getName();
msg.addString(containerName.empty() ? "item of type" + container->getID() : containerName);
```

Comment (Zbizu) [reply src/protocolgame.cpp:1487]: what about this?
`msg.addString(containerName.empty() ? "item of type" + container->getID() : container->getName());`

Comment (Zbizu) [reply src/protocolgame.cpp:1487]: I don't think so
obraz

Comment (DSpeichert) [reply src/protocolgame.cpp:1487]: Shouldn't this be built into `container->getName()` itself?

**Comment by DSpeichert** [comment]
I'd also print a warning to the console.

File: src/protocolgame.cpp (hunk 1 above)
Comment (nekiro) [reply src/protocolgame.cpp:1487]: @DSpeichert this is absolute awesome idea

**Comment by Zbizu** [comment]
@DSpeichert Do we really need to harass server owners with such trivial thing? There are tons of new items that don't spit warnings when they're being interacted with.

File: src/container.cpp [hunk 2]
```diff
@@ -85,6 +85,11 @@ Container* Container::getParentContainer()
 	return thing->getContainer();
 }
 
+std::string Container::getName() const {
```
Comment (nekiro) [inline src/container.cpp:88]: I think we could add bool addArticle here defaulted to false like its done in item class
so std::string Container::getName(bool addArticle /* = false*/) const {
and pass that variable to getNameDescription call

File: src/protocolgame.cpp (hunk 1 above)
Comment (Zbizu) [reply src/protocolgame.cpp]: getName() method added

File: src/container.cpp (hunk 2 above)
Comment (Zbizu) [reply src/container.cpp]: added and tested

**Pull request merged**